crates/chia-protocol/src/wallet_protocol.rs

```diff
@@ -214,3 +214,79 @@ pub struct RequestFeeEstimates {
 pub struct RespondFeeEstimates {
     estimates: FeeEstimateGroup,
 }
+
+#[streamable(message)]
+pub struct RequestRemovePuzzleSubscriptions {
+    puzzle_hashes: Option<Vec<Bytes32>>,
+}
+
+#[streamable(message)]
+pub struct RespondRemovePuzzleSubscriptions {
+    puzzle_hashes: Vec<Bytes32>,
+}
+
+#[streamable(message)]
+pub struct RequestRemoveCoinSubscriptions {
+    coin_ids: Option<Vec<Bytes32>>,
+}
+
+#[streamable(message)]
+pub struct RespondRemoveCoinSubscriptions {
+    coin_ids: Vec<Bytes32>,
+}
+
+#[streamable]
+pub struct CoinStateFilters {
+    include_spent: bool,
+    include_unspent: bool,
+    include_hinted: bool,
+    min_amount: u64,
+}
+
+#[streamable(message)]
+pub struct RequestPuzzleState {
+    puzzle_hashes: Vec<Bytes32>,
+    previous_height: Option<u32>,
+    header_hash: Bytes32,
+    filters: CoinStateFilters,
+    subscribe_when_finished: bool,
+}
+
+#[streamable(message)]
+pub struct RespondPuzzleState {
+    puzzle_hashes: Vec<Bytes32>,
+    height: u32,
+    header_hash: Bytes32,
+    is_finished: bool,
+    coin_states: Vec<CoinState>,
+}
+
+#[streamable(message)]
+pub struct RejectPuzzleState {
+    reason: u8, // RejectStateReason
+}
+
+#[streamable(message)]
+pub struct RequestCoinState {
+    coin_ids: Vec<Bytes32>,
+    previous_height: Option<u32>,
+    header_hash: Bytes32,
+    subscribe: bool,
+}
+
+#[streamable(message)]
+pub struct RespondCoinState {
+    coin_ids: Vec<Bytes32>,
+    coin_states: Vec<CoinState>,
+}
+
+#[streamable(message)]
+pub struct RejectCoinState {
+    reason: u8, // RejectStateReason
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum RejectStateReason {
```

Comment:
don't we support `enum` in the `Streamable` macro? I thought as long as they were simple enums it would work

Reply:
How would the Python bindings work for that?

Reply:
I thought it just turned it into an `int`.

Reply:
I would think you'd want to also add `#[repr(u8)]` here.